iree/compiler/Dialect/Flow/Transforms/ConvertToFlowTensorOps.cpp

```diff
@@ -250,6 +250,32 @@ struct SubTensorToTensorSlice : public OpRewritePattern<SubTensorOp> {
   }
 };
 
+/// Converts linalg.fill ops that are not used by other Linalg ops into
+/// flow.tensor.splat ops.
+///
+/// This is expected to improve performance because we can use DMA
+/// functionalities for the fill, instead of dispatching kernels.
+///
+/// It assumes linalg.fill ops that has linalg op users can be fused with
+/// its users so those cases aren't supported here.
+struct LinalgFillToFlowTensorSplat final
+    : public OpRewritePattern<linalg::FillOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::FillOp fillOp,
+                                PatternRewriter &rewriter) const override {
+    for (Operation *userOp : fillOp->getUsers()) {
```

Comment:
Sorry, follow up from previous comment, We can drop this check if we can run the pass before and after. This check is inherently hard to maintain.